plugins/nf-nomad/src/test/nextflow/nomad/MockScriptRunner.groovy

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2024-2024, Evaluacion y Desarrollo de Negocios, Spain
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nextflow.nomad
+
+import groovy.util.logging.Slf4j
+import groovyx.gpars.dataflow.DataflowBroadcast
+import nextflow.Session
+import nextflow.executor.Executor
+import nextflow.executor.ExecutorFactory
+import nextflow.nomad.executor.NomadExecutor
+import nextflow.processor.TaskHandler
+import nextflow.processor.TaskMonitor
+import nextflow.processor.TaskRun
+import nextflow.processor.TaskStatus
+import nextflow.script.BaseScript
+import nextflow.script.ChannelOut
+import nextflow.script.ScriptRunner
+import nextflow.script.ScriptType
+
+import java.nio.file.Paths
+
+/**
+ * Mock runner for test
+ *
+ * @author Jorge Aguilera <jagedn@gmail.com>
+ */
+
+class MockScriptRunner extends ScriptRunner {
```

Comment:
@jagedn , for future - I think that it'd be great to have some latency option within the mock server given that the real-world cluster network might not be reliable. What do you think?

Reply:
yes, good idea

with mock server we can simulate lot of responses and test all this kind of situations